lib/mirage_logs.ml

```diff
@@ -111,11 +115,16 @@ module Make (C : V1.CLOCK) = struct
     | None -> ()
     | Some _ ->
       let old_hook = !Lwt.async_exception_hook in
+      t.old_hook <- old_hook;
       Lwt.async_exception_hook := (fun ex ->
           dump_ring t t.ch;
           old_hook ex
         )
 
+  let unset_reporter t =
```

Comment:
Calling `unset_reporter` twice (or once without calling set) will cause all further exceptions to be silently discarded. That doesn't seem like a good idea...